agent/lib/kontena/workers/log_worker.rb

```diff
@@ -59,9 +61,17 @@ def start
         end
       end
 
+      async.watch_queue
       async.process_queue
     end
 
+    def watch_queue
+      loop do
+        sleep 10
+        warn "queue size is #{@queue.size}" if @queue.size > (QUEUE_MAX_SIZE / 2)
```

Comment:
have this also mention: "queue size is #{@queue.size}, throttling/dropping container logs"